orm.go

```diff
@@ -138,65 +132,89 @@ func checkTableColumns(tdx Tdx, s interface{}) error {
 	return checkStruct(s, cols, tableName)
 }
 
+type SqlLog struct {
+	Sql      string        `json:"sql"`
+	Duration time.Duration `json:"duration"`
+	Explain  *Explain      `json:"explain,omitempty"`
+}
+type Explain struct {
+	Table  string `json:"table"`
+	Type   string `json:"type"`
+	Key    string `json:"key"`
+	KeyLen int64  `json:"key_len"`
+	Ref    string `json:"ref"`
+	Rows   int64  `json:"rows"`
+	Extra  string `json:"extra"`
+}
+type SqlLogger interface {
+	Log(sqlLog *SqlLog)
+	ShowExplain() bool
+}
+type NoopSqlLogger struct{}
+
+func (n *NoopSqlLogger) Log(sqlLog *SqlLog) {
+	data, _ := json.Marshal(sqlLog)
+	log.Printf("[go-orm] %v\n", string(data))
+}
+func (n *NoopSqlLogger) ShowExplain() bool {
+	return true
+}
+
 func exec(tdx Tdx, query string, args ...interface{}) (sql.Result, error) {
-	if verbose >= LogLevelShowNothing {
-		log.Println("[go-orm] exec sql:", query, args)
-	}
-	return tdx.Exec(query, args...)
+	query = regexp.MustCompile("\\s+").ReplaceAllString(query, " ")
+	start := time.Now()
+	res, err := tdx.Exec(query, args...)
+	defer func() {
+		duration := time.Since(start)
+		sqlLog := SqlLog{Duration: duration, Sql: fmt.Sprintf("%s %v", query, args)}
+		sqlLogger.Log(&sqlLog)
+	}()
```

Comment:
做一个统一的入口：

```
func log(logger SqlLogger, start time.Time, tdx Tdx, query string, args ...interface{}) {
    sqlQuery := regexp.MustCompile("\\s+").ReplaceAllString(query, " ")
    sqlLog := SqlLog{Duration: time.Since(start), Sql: fmt.Sprintf("%s %v", sqlQuery, args)}
    if logger.ShowExplain() {
       // do explain job...
    }
    sqlLogger.Log(&sqlLog)
}

func exec(tdx Tdx, queryStr string, args ...interface{}) (sql.Result, error) {
    defer log(sqlLogger, time.Now(), tdx, queryStr, args...)
    return tdx.Exec(queryStr, args...)
}

func query(tdx Tdx, queryStr string, args ...interface{}) (*sql.Rows, error) {
    defer log(sqlLogger, time.Now(), tdx, queryStr, args...)
    return tdx.Query(queryStr, args...)
}
```